cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -678,7 +678,8 @@ func (s *service) processExits() {
 		// process.
 		var cps, skipped []containerProcess
 		for _, cp := range s.running[e.Pid] {
-			if s.pendingExecs[cp.Container] != 0 {
+			_, init := cp.Process.(*process.Init)
+			if init && s.pendingExecs[cp.Container] != 0 {
```

Comment:
Can we have a test?

Reply:
Yes, I have a patch locally, I'll clean it up and open a PR with the tests.